src/backend/metal/src/command.rs

```diff
@@ -441,7 +443,8 @@ impl RawCommandBuffer<Backend> for CommandBuffer {
         }
         if let EncoderState::Render(ref encoder) = inner.encoder_state {
             for (i, &(buffer, offset)) in buffer_set.0.iter().enumerate() {
-                encoder.set_vertex_buffer(i as _, offset as _, Some(&buffer.0));
+                let msl_buffer_index = inner.attribute_buffer_index as usize + i;
```

Comment:
Probably this offset should be added few lines above too.

Why do we need to store buffers in `resources_vs` anyway?